ql/src/java/org/apache/hadoop/hive/ql/parse/ParseUtils.java

```diff
@@ -565,8 +565,14 @@ public static Map<Integer, List<ExprNodeGenericFuncDesc>> getFullPartitionSpecs(
       CommonTree ast, Table table, Configuration conf, boolean canGroupExprs) throws SemanticException {
     String defaultPartitionName = HiveConf.getVar(conf, HiveConf.ConfVars.DEFAULTPARTITIONNAME);
     Map<String, String> colTypes = new HashMap<>();
-    for (FieldSchema fs : table.getPartitionKeys()) {
-      colTypes.put(fs.getName().toLowerCase(), fs.getType());
+    if (table.getStorageHandler() != null && table.getStorageHandler().alwaysUnpartitioned()) {
+      for (FieldSchema fs : table.getStorageHandler().getPartitionKeys(table)) {
```

Comment:
should we extract population of partition keys

List<FieldSchema> pk =  table.getStorageHandler() != null && table.getStorageHandler().alwaysUnpartitioned()?table.getStorageHandler().getPartitionKeys(table) : table.getPartitionKeys()

Reply:
Done.